atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1223,18 +1278,44 @@ private static boolean isSweepSentinel(Value value) {
      * This will return all the keys that still need to be postFiltered.  It will output properly
      * postFiltered keys to the results output param.
      */
-    private <T> Map<Cell, Value> getWithPostFilteringInternal(TableReference tableRef,
+    private <T> ListenableFuture<Map<Cell, Value>> getWithPostFilteringInternal(
+            TableReference tableRef,
             Map<Cell, Value> rawResults,
             @Output ImmutableMap.Builder<Cell, T> results,
             @Output AtomicInteger count,
-            Function<Value, T> transformer) {
-        Set<Long> startTimestampsForValues = getStartTimestampsForValues(rawResults.values());
-        Map<Long, Long> commitTimestamps = getCommitTimestamps(tableRef, startTimestampsForValues, true);
+            Function<Value, T> transformer,
+            CellLoader cellLoader) {
+
+        Set<Cell> orphanedSentinels = findOrphanedSweepSentinels(tableRef, rawResults);
+
+        return Futures.transformAsync(
+                getCommitTimestamps(tableRef, getStartTimestampsForValues(rawResults.values()), true, cellLoader),
```

Comment:
uninline the `getStartTimestamps` please

Reply:
Changed.

Reply:
Refactored.